fuzz/fuzz_targets/refcounted_model.rs

```diff
@@ -52,10 +52,7 @@ impl DbSimulator for Simulator {
 			Operation::Dereference(k) =>
 				if let Entry::Occupied(mut e) = model.entry(k) {
 					let counter = e.get_mut();
-					*counter -= 1;
-					if *counter == 0 {
-						e.remove_entry();
-					}
+					*counter = counter.saturating_sub(1);
```

Comment:
Reading how dereference work (in write_dec_ref), when reaching 0 next calls to Reference should be noops, so I am not sure why we did this change.

Reply:
Oh, is it because the fuzzer might get its data from the commit overlay?
Ok I start to understand :facepalm: 
Indeed it's not trivial to know where the data is, so I get why we check in a less restrictive way.
I would think this 'optional model' less restrictive check should be only for ref counted and not in other case.

Reply:
This change is here because of two mistakes I made:
- Indeed, extra dereferences after reaching 0 are no op and not doing overflow as the current (i.e. before this PR) fuzzer model is doing.
- The database is not immediately garbage collecting the value when the counter reaches 0 so the value should still be stored in the model because database iteration might return it.

Reply:
Yes I did not get it at first, but make sense that we cannot do some very complete checks when using ref_counted (due to how things behave in the commit overlay).
Maybe using the optional model only when using ref_counted and keep an exhaustive check for btree without refcounted value can be a small improvment. (Simply having the optional model function return an optional value).

Reply:
>  Maybe using the optional model only when using ref_counted and keep an exhaustive check for btree without refcounted value can be a small improvment. (Simply having the optional model function return an optional value).

Yes, it's what is done now in PR after your comment about checking removed values. In `simple_model` removed keys are now checked for non-existence in the database.